php/classes/controllers/class-players-controller.php

```diff
@@ -0,0 +1,181 @@
+	 */
+	public function get_feed_url() {
+		// Get feed slug
+		$feed_slug = apply_filters( 'ssp_feed_slug', $this->token );
+
+		if ( get_option( 'permalink_structure' ) ) {
+			$feed_url = $this->home_url . 'feed/' . $feed_slug;
+		} else {
+			$feed_url = $this->home_url . '?feed=' . $feed_slug;
+		}
+
+		$custom_feed_url = get_option( 'ss_podcasting_feed_url' );
+		if ( $custom_feed_url ) {
+			$feed_url = $custom_feed_url;
+		}
+
+		$feed_url = apply_filters( 'ssp_feed_url', $feed_url );
+
+		return $feed_url;
+	}
+
+	/**
+	 * Return html player for a given podcast (episode) id.
+	 *
+	 * @param int $id
+	 *
+	 * @return string
+	 */
+	public function html_player( $id ) {
+		$episode         = get_post( $id );
+		$episodeDuration = get_post_meta( $id, 'duration', true );
+		$audioFile       = get_post_meta( $id, 'audio_file', true );
+		$albumArt        = $this->episode_controller->get_album_art( $id );
+		$podcastTitle    = get_option( 'ss_podcasting_data_title' );
+
+		$subscribeLinks = $this->subscribe_links( $id );
+
+		$feedUrl = $this->get_feed_url();
+		// set any other info
+		$templateData = array(
+			'episode'      => $episode,
+			'duration'     => $episodeDuration,
+			'audioFile'    => $audioFile,
+			'albumArt'     => $albumArt,
+			'podcastTitle' => $podcastTitle,
+			'feedUrl'      => $feedUrl,
+			'itunes'       => $subscribeLinks['itunes'],
+			'stitcher'     => $subscribeLinks['stitcher'],
+			'spotify'      => $subscribeLinks['spotify'],
+			'googlePlay'   => $subscribeLinks['googlePlay']
+		);
+
+		$templateData = apply_filters( 'ssp_html_player_data', $templateData );
+
+		return $templateData;
+
+	}
+
+	/**
+	 * Return media player for a given podcast (episode) id.
+	 *
+	 * @param int $id
+	 *
+	 * @return string
+	 */
+	public function media_player( $id ) {
+		// get src file
+		$srcFile = get_post_meta( $id, 'audio_file', true );
+		$params  = array(
+			'src'     => $srcFile,
+			'preload' => 'none'
+		);
+
+		$mediaPlayer = wp_audio_shortcode( $params );
+
+		return $mediaPlayer;
+	}
+
+	public function subscribe_links( $id ) {
+
+		$seriesId = $this->get_series_id( $id );
+
+		if ( $seriesId ) {
+			$itunes     = get_option( "ss_podcasting_itunes_url_{$seriesId}" );
+			$stitcher   = get_option( "ss_podcasting_stitcher_url_{$seriesId}" );
+			$spotify    = get_option( "ss_podcasting_spotify_url_{$seriesId}" );
+			$googlePlay = get_option( "ss_podcasting_google_play_url_{$seriesId}" );
+		} else {
+			$itunes     = get_option( "ss_podcasting_itunes_url" );
+			$stitcher   = get_option( "ss_podcasting_stitcher_url" );
+			$spotify    = get_option( "ss_podcasting_spotify_url" );
+			$googlePlay = get_option( "ss_podcasting_google_play_url" );
+		}
+
+		$subscribeLinks = array(
+			'itunes'     => $itunes,
+			'stitcher'   => $stitcher,
+			'spotify'    => $spotify,
+			'googlePlay' => $googlePlay
+		);
+
+		return $subscribeLinks;
+	}
+
+	public function get_series_id( $episode_id ) {
+		$series_id = false;
```

Comment:
Can we default this to 0 instead of false. That way the return from the method is always an integer